components/pages/library/CollectionData.js

```diff
@@ -0,0 +1,57 @@
+import { useState, useEffect, useContext } from "react";
+import Link from "next/link";
+
+import CoverFromData from "@/components/common/Cover/coverData.js";
+
+import { genreData } from "@/public/data/genre.js";
+
+import { DataContext } from "@/contexts/dataContext.js";
+
+import styles from "./collection.module.css";
+
+export default function CollectionFromData() {
+  const { bookData } = useContext(DataContext);
+  const [isLoading, setIsLoading] = useState(true);
+
+  const FilterData = (array, key, value) => {
+    return array.filter((item) => item[key] === value);
+  };
+
+  useEffect(() => {
+    if (bookData && bookData.length > 0) {
+      setIsLoading(false);
+    }
+  }, [bookData]);
+
+  function render(filter) {
+    const filteredArray = FilterData(bookData, "genre", filter);
+    let content = [];
+    for (let i = 0; i < filteredArray.length; i++) {
+      content.push(
+        <li key={i} className={styles.bookitem}>
+          <Link href={`/library/book/${filteredArray[i].slug}`}>
+            <CoverFromData
+              slug={filteredArray[i].slug}
+              height={220}
+            ></CoverFromData>
+          </Link>
+        </li>
+      );
+    }
```

Comment:
You could use a map at this point.
And the `<CoverFormData>` component seems to be a self-closing one. You don't give it children.
```js
const content = filteredArray.map((item, index) => {
    return (
    <li key={index}>
          <Link href={`/library/book/${filteredArray[index].slug}`}>
            <CoverFromData
              slug={filteredArray[index].slug}
              height={220}
            />
          </Link>
    </li>
    )
}
```